src/commands/github/actionCommands.ts

```diff
@@ -34,4 +40,62 @@ export async function cancelAction(context: IActionContext, node?: ActionTreeIte
     const client: Octokit = await createOctokitClient();
     await client.actions.cancelWorkflowRun({ owner: node.data.repository.owner.login, repo: node.data.repository.name, run_id: node.data.id });
     await node.refresh(); // need to refresh to update the data
+    await checkActionStatus(context, node, 'cancel');
+}
+
+async function checkActionStatus(context: IActionContext, node: ActionTreeItem, action: 'rerun' | 'cancel'): Promise<void> {
+    // realistically, we have no idea how long someone's build can take, but might as well keep retrying while VS Code is opened
+    const tokenSource: CancellationTokenSource = new CancellationTokenSource();
+    const token: CancellationToken = tokenSource.token;
+    if (activeActionTokens[node.data.id]) {
+        activeActionTokens[node.data.id]?.cancel();
+    }
+
+    activeActionTokens[node.data.id] = tokenSource;
+
+    const retries: number = 10;
+    const startTime: number = Date.now();
+    const notCompleteError: string = localize('notComplete', 'Action "{0}" has not completed.', node.data.id);
+    const cancelled: string = 'Action cancelled by user';
+    try {
+        await retry(
+            async () => {
+                if (token?.isCancellationRequested) {
+                    throw new retry.AbortError(cancelled);
+                }
+
+                const client: Octokit = await createOctokitClient();
+                const workflowRun: ActionsGetWorkflowRunResponseData = (await client.actions.getWorkflowRun({ owner: node.data.repository.owner.login, repo: node.data.repository.name, run_id: node.data.id })).data;
+                if (workflowRun.status === 'completed') {
+                    const actionCompleted: string = localize('actionCompleted', 'Action "{0}" has completed with the conclusion "{1}".', node.data.id, workflowRun.conclusion);
+                    ext.outputChannel.appendLog(actionCompleted);
+                    window.showInformationMessage(actionCompleted);
+                    await node.refresh();
+                    context.telemetry.properties.secToReport = String((Date.now() - startTime) / 1000);
+                    context.telemetry.properties.conclusion = workflowRun.conclusion;
+                    context.telemetry.properties.action = action;
+                    return;
+                }
+
+                throw new Error(notCompleteError);
+            },
+            {
+                onFailedAttempt: error => {
+                    if (parseError(error).message !== notCompleteError) {
+                        throw error;
+                    }
+                },
+                retries, minTimeout: 5 * 1000,
+            });
+    } catch (error) {
+        if (parseError(error).message === cancelled) {
```

Comment:
For some reason, the abort error doesn't keep its typing in the catch block, so I have to check the message instead (which is never displayed to the user)